File: go/cars-assemble/cars_assemble.go
package cars

const (
	groupsOfTenPrice = 95000
	individualPrice  = 10000
)

// CalculateWorkingCarsPerHour calculates how many working cars are
// produced by the assembly line every hour
func CalculateWorkingCarsPerHour(productionRate int, successRate float64) float64 {
	rate := float64(productionRate) / 100.0
	return rate * successRate
}

// CalculateWorkingCarsPerMinute calculates how many working cars are
// produced by the assembly line every minute
func CalculateWorkingCarsPerMinute(productionRate int, successRate float64) int {
	carsPerHour := CalculateWorkingCarsPerHour(productionRate, successRate)
	carsPerMinute := int(carsPerHour) / 60
	return carsPerMinute
}

// CalculateCost works out the cost of producing the given number of cars
func CalculateCost(carsCount int) uint {
	groupsOfTen := carsCount / 10
	remainder := carsCount % 10
	cost := uint(groupsOfTen) * groupsOfTenPrice
	cost += uint(remainder) * individualPrice
	return cost
}

File: clojure/cars-assemble/src/cars_assemble.clj
(ns cars-assemble)

(def cars-per-hour 221)

(defn success-rate [speed]
  (cond (> speed 9) 0.77
        (> speed 8) 0.8
        (> speed 4) 0.9
        (> speed 0) 1.0
        :else 0.0))

(defn production-rate
  "Returns the assembly line's production rate per hour,
   taking into account its success rate"
  [speed]
  (* cars-per-hour speed (success-rate speed)))

(defn working-items
  "Calculates how many working cars are produced per minute"
  [speed]
  (int (/ (production-rate speed) 60)))
